consumer.go

```diff
@@ -93,20 +95,50 @@ func NewConsumerForRemote(ctx context.Context, remote *bfs.Object, opt *Consumer
 		cfn:    cfn,
 	}
 
-	// run initial sync
-	if _, err := c.sync(true); err != nil {
-		_ = c.Close()
+	return c.run()
+}
+
+// NewIncrementalConsumer starts a new incremental feed consumer.
+func NewIncrementalConsumer(ctx context.Context, bucketURL string, opt *ConsumerOptions, cfn ConsumeFunc) (Consumer, error) {
```

Comment:
other option is to have single constructor func and pass `s3://bucket/path/manifest.json` as remoteURL (and determine bucket from URL) and set `Incremental: true` in opts...  however I also quite like having an explicit NewIncremental constructor.